src/dsp_tools/commands/xmlupload/models/ingest.py

```diff
@@ -0,0 +1,86 @@
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Protocol
+
+import requests
+from loguru import logger
+from requests import Session
+from requests.adapters import HTTPAdapter
+from requests.adapters import Retry
+
+from dsp_tools.models.exceptions import BadCredentialsError
+from dsp_tools.models.exceptions import PermanentConnectionError
+from dsp_tools.utils.logger_config import logger_savepath
+
```

Comment:
Are all of these imports necessary?

Reply:
ruff (our linter) warns us when there are unused imports, so they are certainly used, yes